### PR TITLE
Improve HDYC Dark Mode

### PR DESCRIPTION
**Pull request by Nekzuris**
I fixed a few things and improve contrast for better readability.

### CODE REVIEW

**Comment by deevroman** [comment]
What do you fix with?
```
body {
	overflow-x: auto !important;
}
```

It looks good, and I'll merge it in master a little later to release additional changes so as not to distract users with updates.

**Comment by Nekzuris** [comment]
There is an unnecessary scroll bar at the bottom of the iframe, less noticeable on Chrome because it's at the very bottom, not sure where it comes from.
image
Btw the height of the iframe should be based on the content but idk it doesn't seem to work.
And I've put it outside the media query for light mode but it's still useless because the whole stylesheet is only added if dark mode is detected, maybe just remove the `isDarkMode()` check?

**Comment by deevroman** [comment]
The height is difficult. HDYC is inserted into the page as an Iframe and I didn't find a way other than to fix its height so that when scrolling, the entire page scrolls, and not just the content of the iframe

>  remove the isDarkMode() check

Yes, it seems `@media (preferences-color-scheme: dark) {` is enough, and I can always add a style

**Comment by Nekzuris** [comment]
I improved the selectors and changed how the map filter is applied, still looks the same but cleaner CSS.

**Comment by Nekzuris** [comment]
Also imo you shouldn't remove the mapper_img, especially if the user has a long profile description, I don't think it's a problem to see it twice.

File: better-osm-org.user.js [hunk 1]
```diff
@@ -6151,89 +6151,77 @@ function simplifyHDCYIframe() {
...
-                i.style.color = "lightgray";
-            }
-        })
-        document.querySelectorAll("td").forEach(i => {
-            if (i.style.color === "purple") {
-                i.style.color = "orchid";
             }
-        })
-    }
-    document.querySelector("html").style.overflowX = "scroll"
```
[the remark this answers is not in the document]
Comment (deevroman) [reply better-osm-org.user.js:6200]: checked, everything is ok

**Review by deevroman — APPROVED** [review]

**Comment by deevroman** [comment]
> Also imo you shouldn't remove the mapper_img, especially if the user has a long profile description, I don't think it's a problem to see it twice.

I'll think about it, but the original idea was to reduce duplication. Still, this is a fairly large element that takes attention. Another possible side effect is that in theory it can save space on the phone screen (but it may not work right now)

**Comment by Nekzuris** [comment]
Feels empty
image
It's not a big image, on forum threads you see the profile picture next to every post and it doesn't feel wrong.

**Pull request merged**